### PR TITLE
DOC: add deprecation message to index_update and friends

### LINKED ISSUE #8064: Update jax.ops.index_* documentation to include current preferred index notation
If you search on Google for how to update array indices in JAX, you end up on [this documentation page](https://jax.readthedocs.io/en/latest/_autosummary/jax.ops.index_update.html#jax-ops-index-update) for `jax.ops.index_update`.

`jax.ops.index_update` is no longer the best way to update indices -- in most cases users should instead write code like `x.at[idx].set(y)`, which is a really great improvement!

The documentation for the `jax.ops.index_*` should also describe and/or link to the alternate, simpler, and preferred notation for working with indices.

### PR DESCRIPTION
**Pull request by jakevdp**
Fixes #8064

Preview the new docs here: https://jax--8065.org.readthedocs.build/en/8065/_autosummary/jax.ops.index_update.html#jax.ops.index_update

### CODE REVIEW

**Comment by hawkinsp** [comment]
This looks good.

There is one further step we probably should do. The `index_mul` etc docs have an additional and necessary paragraph:

"
Unlike the NumPy code x[idx] *= y, if multiple indices refer to the same location the updates will be multiplied. (NumPy would only apply the last update, rather than multiplying the updates.) The order in which conflicting updates are applied is implementation-defined and may be nondeterministic (e.g., due to concurrency on some hardware platforms).
"

We should probaby add that to the `.at[...]` documentation, suitably modified.

**Review by hawkinsp — APPROVED** [review]

**Comment by hawkinsp** [comment]
> This looks good.
> 
> There is one further step we probably should do. The `index_mul` etc docs have an additional and necessary paragraph:
> 
> " Unlike the NumPy code x[idx] *= y, if multiple indices refer to the same location the updates will be multiplied. (NumPy would only apply the last update, rather than multiplying the updates.) The order in which conflicting updates are applied is implementation-defined and may be nondeterministic (e.g., due to concurrency on some hardware platforms). "
> 
> We should probaby add that to the `.at[...]` documentation, suitably modified.

There is also some useful parameter documentation on those functions that needs to be transferred. Maybe for a future PR, though.

**Comment by jakevdp** [comment]
Added the extra paragraph, as well as a TODO to document additional keyword arguments.

File: jax/_src/numpy/lax_numpy.py [hunk 1]
```diff
@@ -6690,22 +6690,43 @@ class _IndexUpdateHelper:
...
+  ``x = x.at[idx].get()``         ``x = x[idx]``
+  ==============================  ================================
+
+  None of these expressions modify the original ``x``; instead they return
+  a modified copy of ``x``. However, inside a :py:func:`jax.jit` compiled function,
+  expressions like ``x = x.at[idx].set(y)`` are guaranteed to be applied in-place.
+
+  Unlike NumPy in-place operations such as :code:`x[idx] += y`, if multiple
+  indices refer to the same location, all updates will be applied (NumPy would
+  only apply the last update, rather than multiplying the updates.) The order
```
Comment (hawkinsp) [inline jax/_src/numpy/lax_numpy.py:6716]: "multiplying" only makes sense for `index_mul` perhaps. But instead "applying" might work.

**Review by hawkinsp — APPROVED** [review]

File: jax/_src/numpy/lax_numpy.py (hunk 1 above)
Comment (jakevdp) [reply jax/_src/numpy/lax_numpy.py]: Done.

**Pull request merged**